libs/s25main/ai/aijh/AIPlayerJH.cpp

```diff
@@ -1704,7 +1702,8 @@ void AIPlayerJH::TrySeaAttack()
     unsigned limit = 15;
     unsigned skip = 0;
     if(searcharoundharborspots.size() > 15)
-        skip = std::max<int>(rand() % (searcharoundharborspots.size() / 15 + 1) * 15, 1) - 1;
+        skip =
+          std::max<int>(AI::randomValue<int>(0, static_cast<int>(searcharoundharborspots.size() / 15) * 15), 1) - 1;
```

Comment:
This isn't the same.
16: -> `std::max<int>(rand(0,1) * 15, 1) - 1; -> 0 or 14`
30: -> `std::max<int>(rand(0,2) * 15, 1) - 1; -> 0, 14, 29`

The logic doesn't seem to make sense to me. It seems it wants one of the chunks of size 15, i.e. `[0,14)`, `[15, 29)` etc.

Maybe this is intended:
```suggestion
        skip = AI::randomValue(0u, searcharoundharborspots.size() / 15u) * 15u;
```
so for 16-29 we get either 0 or 15 and for 30-44 we get 0, 15, or 30